src/main/java/com/juniorstart/juniorstart/service/GoalService.java

```diff
@@ -0,0 +1,130 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.ResourceNotFoundException;
+import com.juniorstart.juniorstart.exception.SaveException;
+import com.juniorstart.juniorstart.model.Goal;
+import com.juniorstart.juniorstart.model.User;
+import com.juniorstart.juniorstart.payload.ApiResponse;
+import com.juniorstart.juniorstart.payload.GoalRequestUpdate;
+import com.juniorstart.juniorstart.repository.GoalDao;
+import com.juniorstart.juniorstart.repository.UserDao;
+import com.juniorstart.juniorstart.util.ClassMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/** Represents an Goal service.
+ * @author Dawid Wit
+ * @version 1.0
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class GoalService {
+
+    private final GoalDao goalDao;
+    private final UserDao userDao;
+
+    /** Create new goal for User.
+     * @param email of logged user.
+     * @param password of logged user.
+     * @param dto of new goal.
+     * @return ResponseEntity<ApiResponse> is goal had been created.
+     * @throws ResourceNotFoundException if user is not found.
+     * @throws SaveException if GoalDao save error.
+     */
+    public ResponseEntity<ApiResponse> create(String email, String password, Goal.GoalDto dto) {
+        User user = findUser(email, password);
+        Goal goal = ClassMapper.mapToEntity(dto);
+        goal.setUser(user);
+
+        if (isDuplicatedGoal(dto.getName(), user)){
+            return ResponseEntity.ok(new ApiResponse(false,"Goal duplicate name"));
```

Comment:
Ja gdzieś natrafiłem informację, żeby zwracać 409, ale niech ktoś inni się wypowie.
Bynajmniej 200  na pewno nie wydaje się ok

Reply:
ja bym rzucił BadRequestem, ale to jest moje zdanie

Reply:
Dalej nie zmienione 👍